packages/core/src/editor/actions.ts

```diff
@@ -150,48 +171,90 @@ export const Actions = (
 
       currentParentNodes[currentParentNodes.indexOf(targetId)] = "marked";
 
-      if (newParentNodes) newParentNodes.splice(index, 0, targetId);
-      else newParent.data.nodes = [targetId];
+      if (newParentNodes) {
+        newParentNodes.splice(index, 0, targetId);
+      } else {
+        newParent.data.nodes = [targetId];
+      }
 
       state.nodes[targetId].data.parent = newParentId;
       state.nodes[targetId].data.index = index;
       currentParentNodes.splice(currentParentNodes.indexOf("marked"), 1);
     },
+
+    replaceEvents(events: EditorEvents) {
```

Comment:
Is there a use case for manually manipulating editor events (as oppose to using `setNodeEvent` and `setIndicator`) ?  Otherwise, I would combine this into `reset()` where we simply reset the editor events.